backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inout/invoicecandidate/M_InOutLine_Handler.java

```diff
@@ -823,7 +822,7 @@ public static PriceAndTax calculatePriceAndTax(
 				.taxCategoryId(pricingResult.getTaxCategoryId())
 				//
 				.priceEntered(pricingResult.getPriceStd())
-				.priceActual(pricingResult.getPriceStd())
+				.priceActual(inoutLine.getC_OrderLine_ID() != 0 ? Services.get(IOrderLineBL.class).getPriceActual(inoutLine.getC_OrderLine()) : pricingResult.getPriceStd())
```

Comment:
we can'T combine a number from a C_OrderLine with an UOM from the prcing engine. there is no guarantee that they fit together

Reply:
removed old logic, using the existing getcostprice() method from orderlineBL